lib/src/selectors/time_picker.dart

```diff
@@ -0,0 +1,403 @@
+///
+/// The [onTimeChanged] callback passes through the user's selected time, and
+/// must be provided.
+/// {endtemplate}
+class MacosTimePicker extends StatefulWidget {
+  /// {@macro macosTimePicker}
+  const MacosTimePicker({
+    Key? key,
+    required this.onTimeChanged,
+    this.style = TimePickerStyle.combined,
+  }) : super(key: key);
+
+  /// The [TimePickerStyle] to use.
+  ///
+  /// Defaults to [TimePickerStyle.combined].
+  final TimePickerStyle style;
+
+  /// {@macro onTimeChanged}
+  final OnTimeChanged onTimeChanged;
+
+  @override
+  State<MacosTimePicker> createState() => _MacosTimePickerState();
+}
+
+class _MacosTimePickerState extends State<MacosTimePicker> {
+  final _initialTime = TimeOfDay.now();
+  late int _selectedHour;
+  late int _selectedMinute;
+  late DayPeriod _selectedPeriod;
+  bool _isHourSelected = false;
+  bool _isMinuteSelected = false;
+  bool _isPeriodSelected = false;
+  late MaterialLocalizations localizations = MaterialLocalizations.of(context);
+  late var time = DateTime(
+    DateTime.now().year,
+    DateTime.now().month,
+    DateTime.now().day,
+    _selectedHour,
+    _selectedMinute,
+    DateTime.now().second,
+  );
+
+  @override
+  void initState() {
+    super.initState();
+    _parseInitialTime();
+    if (widget.style == TimePickerStyle.graphical ||
+        widget.style == TimePickerStyle.combined) {
+      Timer.periodic(const Duration(seconds: 1), update);
+    }
+  }
+
+  void update(Timer timer) {
+    if (mounted) {
+      time = time.add(const Duration(seconds: 1) * timer.tick);
+      setState(() {});
+    }
+  }
+
+  void _parseInitialTime() {
+    _selectedHour = _initialTime.hour;
+    _selectedMinute = _initialTime.minute;
+    _selectedPeriod = _initialTime.period;
+  }
+
+  TimeOfDay _formatAsTimeOfDay() {
+    return TimeOfDay(hour: _selectedHour, minute: _selectedMinute);
+  }
+
+  void _incrementElement() {
+    if (_isHourSelected) {
+      if (_selectedHour + 1 >= 24) {
+        setState(() {
+          _selectedHour = 0;
+          _selectedPeriod = DayPeriod.am;
+        });
+      } else {
+        if (_selectedHour + 1 >= 12) {
+          setState(() {
+            _selectedHour++;
+            _selectedPeriod = DayPeriod.pm;
+          });
+        } else {
+          setState(() => _selectedHour++);
+        }
+      }
+    } else if (_isMinuteSelected) {
+      if (_selectedMinute + 1 > 59) {
+        setState(() => _selectedMinute = 0);
+      } else {
+        setState(() => _selectedMinute++);
+      }
+    } else {
+      if (_selectedPeriod == DayPeriod.am) {
+        setState(() {
+          _selectedHour = _selectedHour + 12;
+          _selectedPeriod = DayPeriod.pm;
+        });
+      } else {
+        setState(() {
+          _selectedHour = _selectedHour - 12;
+          _selectedPeriod = DayPeriod.am;
+        });
+      }
+    }
+    widget.onTimeChanged.call(_formatAsTimeOfDay());
+  }
+
+  void _decrementElement() {
```

Comment:
Same as above about using a single setState((){}) call.

Reply:
AFAIK doing conditional statement inside setState is not recommended.